controllers/bootstrap/init.go

```diff
@@ -717,9 +717,13 @@ func (b *Bootstrap) CreateOrUpdateFromYaml(yamlContent []byte, alwaysUpdate ...b
 
 		// do not compareVersion if the resource is subscription
 		if gvk.Kind == "Subscription" {
-			sub, err := b.GetSubscription(ctx, obj.GetName(), b.CSData.MasterNs)
+			sub, err := b.GetSubscription(ctx, obj.GetName(), obj.GetNamespace())
 			if err != nil {
-				klog.Errorf("Failed to get subscription %s/%s", b.CSData.MasterNs, obj.GetName())
+				if obj.GetNamespace() == "" {
+					klog.Errorf("Failed to get subscription for %s. Namespace not found.", obj.GetName())
+				} else {
+					klog.Errorf("Failed to get subscription %s/%s", obj.GetNamespace(), obj.GetName())
+				}
 			}
```

Comment:
Hi @bluzarraga 

When the `err != nil` happens, the `sub` would be `nil`.
Therefore, if it continues to run into `update = !equality.Semantic.DeepEqual(sub.Object["spec"], obj.Object["spec"])`, we will hit the `nil` pointer issue for `sub` and crash the pod. Therefore, we should `return err` directly.

Reply:
Also, I think the additional check for `obj.GetNamespace() == ""`  is good. It will help to provide more information if the error happens.

Reply:
Ok I see what you're saying now, the nil pointer is from the deep compare